src/nvim/ex_cmds2.c

```diff
@@ -3403,16 +3413,20 @@ static void script_host_execute_file(char *name, exarg_T *eap)
   // filename
   list_append_string(args, buffer, -1);
   // current range
-  list_append_number(args, eap->line1);
-  list_append_number(args, eap->line2);
+  assert(eap->line1 <= INT_MAX);
+  assert(eap->line2 <= INT_MAX);
+  list_append_number(args, (int)eap->line1);
+  list_append_number(args, (int)eap->line2);
```

Comment:
your points in #4076 seem reasonable but maybe rename or break out the `linenr_T` part to a separate issue/PR because changing `linenr_T` seems like a bigger change/concern than fixing conversion in `diff.c`